sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetIOSuite.scala

```diff
@@ -1060,6 +1061,25 @@ class ParquetIOSuite extends QueryTest with ParquetTest with SharedSparkSession
       }
     }
   }
+
+  test("SPARK-36825: year-month/day-time intervals written and read as INT32/INT64") {
```

Comment:
Should we care both V1 and V2 Parquet sources?

Reply:
I care of the default case in this PR when users save/load dataframes with ANSI intervals. Let's check/test V2 separately since it might require additional work (this PR cares mostly about low level stuff).